sdlib/d/gc/region.d

```diff
@@ -361,7 +395,30 @@ public:
 
 		import d.gc.util;
 		auto a = min(address, r.address);
-		return at(a, size + r.size);
+
+		// Dirt is at all times contiguous within a region, and starts at the bottom.
+		// Given as purging is not yet supported, this invariant always holds.
+		assert(
+			!(address == a ? &this : r).hasClean
+				|| !(address == a ? r : &this).hasDirt,
```

Comment:
Note that this is actually more work than the proposed code, because both address could be equal (it doesn't happen in practice, but the compiler can't know that).